jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -801,3 +801,32 @@ def _register_spectrum(self, event):
 
         self.add_results.add_results_from_plugin(spectrum)
         self._set_default_results_label()
+
+    def _apply_subset_masks(self, spectrum):
+        """
+        Iterate over dimensions of a spectral cube ``spectrum``, add a mask attribute
+        if none exists. Mask excludes non-selected spectral and/or spatial subsets.
+        """
+        for subset_component in [self.spatial_subset, self.spectral_subset]:
+            if subset_component is None:
+                # spatial_subset only exists for cubeviz
+                continue
+
+            # only look for a mask if there is a selected subset:
+            if subset_component.selected == subset_component.default_text:
+                continue
+
+            subset_mask = subset_component.selected_subset_mask
+
+            if subset_mask.ndim == 1:
```

Comment:
```suggestion
            if subset_mask.ndim == 1 and spectrum.flux.ndim == 3:
```
This fixes the error I encountered.